### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by JawharBairakdar**
# Don't Merge yet
##### This PR is about fixing:

# Done
 - [x] 404 Page NotFound.
 - [x] the Routes flixibility system.
#### Done offline
 - [x] setting up a tiny cache container for increase the number of requests in timelineStore.
 - [x] fixing re-assigning the teachers for each module
 - [x] fixing the archive Button for update the component without any refresh.
 - [x] making the timeline details on the whole app level; for multy use cases.
 - [x] feature: Grant the students to see there teachers.
 - [x] removing useless Context API
 - [x] Adding the needed flixibillity on the timeline details insuring it will fit everywhere.
 - [x] fixing today marker is null if no timeline items provided.
# In Progress
 - [ ] can't find a teacher after it assigned! (assign one you have to deselect the module to find it there)
 # Nextup
 - [ ] protecting the backend users edit somehow.
- [ ] protecting the edit button.
- [ ] protecting the sync button. 
- [ ] Cleaning Up for pushing.
- [ ] clearifing by comments.

### CODE REVIEW

**Comment by JawharBairakdar** [comment]
I couldn't Fix the refresh bug @M3kH

File: src/App.js [hunk 1]
```diff
@@ -205,15 +279,23 @@ class App extends Component {
...
-                                <Switch>
-                                    { // every Route the user has an access scope to it will be rendered here
-                                        securedRouteProps.map(item => <Route key={item.path} {...item} />)
-                                    }
-                                    <Redirect exact strict from='/' to='/timeline' />
-                                    { // the user isn't logged in OR the lifecycle done AND No route found render this Component
-                                        (!this.state.isLoggedIn || this.state.done) && <Route {...this.routes.NotFound} />
-                                    }
-                                </Switch>
+                                {/* <pre>{JSON.stringify(this.state, null, 2)}</pre> */}
```
Comment (JawharBairakdar) [inline src/App.js]: - [ ] Oops!

File: src/components/Header/Header.js [hunk 2]
```diff
@@ -92,7 +92,7 @@ export default class Header extends Component {
                             <li>
                                 {   // if there is a label the page will shows up here
                                     routes.filter(item => item.label).map(route => (
-                                        <NavLink key={route.name} exact to={route.path}>
+                                        <NavLink key={route.label} exact to={route.path}>{/*Bug Fix*/}
```
Comment (JawharBairakdar) [inline src/components/Header/Header.js]: - [ ] Oops!

**Comment by remarcmij** [comment]
Hi @JawharBairakdar, thanks for your PR. With the class 13 project now started on Hyfer we are now working off a new branch. We will review your PR at some point, and may decide to adopt portions of it if it fits in with the new project.

**Comment by JawharBairakdar** [comment]
Hi @remarcmij glad to see your review and adopting promise 🙂. it would be an honor to see that my PR portions of it fits within the new project, But in all cases i am working to shapin my skills and to make the work done 😃.
if there will be a thing can help I have to other than sharping my skills please let me know 🙂.
excited to hear from the review...

**Comment by JawharBairakdar** [comment]
Hi @remarcmij,
I am sorry 😅 I Closed my PR because i get far away of the project it self, so it wold be somehow conflicting to the current master more than fixing in case you want the same working here I will make a new branch for my new changes.
I have played around alot so a new branch would be a nice idea thing for my latest changes.

Thanks a lot 🙂.

**Comment by remarcmij** [comment]
Hi @JawharBairakdar, this week class13 will graduate. We have been working on a `class13` branch throughout this project, which I will merge back to `master` after graduation. Class 13 did not work in the Train Ticket and Homework functionality the class 12 worked on. The main work of class 12 is on replacing the README for students and teachers and the GitHub sync function. Also, the stores based on the observer pattern have been replaced by Mobx stores.

If you want to check whether your PR is still relevant I invite you to do so after this Sunday when the `master` branch has been updated.

**Pull request closed without merging**